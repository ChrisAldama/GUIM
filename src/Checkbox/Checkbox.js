import React, { Component } from 'react';

const Checkbox = props => (
  <input
    type="checkbox"
    name={props.name}
    onChange={props.onChange}
    checked={props.checked} />
);

export default Checkbox;
