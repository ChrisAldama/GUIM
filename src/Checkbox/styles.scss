@import '../styles/colors.scss';

.GUIMCheckbox {
  -webkit-appearance: none;
  background-color: var(--white);
  border: 2px solid var(--medium-gray);
  border-radius: 3px;
  display: inline-block;
  position: relative;
  outline: none;
  width: 16px;
  height: 16px;
  transition: all 0.5s ease;
  cursor: pointer;
}
.GUIMCheckbox:checked{
  background-color: var(--main-blue);
  border: 2px solid var(--main-blue);
  &::before{
    -webkit-transform: rotate(45deg);
    transform: rotate(45deg);
    content: "";
    position: absolute;
    top: -1px;
    left: 4px;
    display: table;
    width: 2px;
    height: 10px;
    border: 2px solid #fff;
    border-top-width: 0;
    border-left-width: 0;
  }
}
