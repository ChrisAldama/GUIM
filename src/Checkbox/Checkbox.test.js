import React from 'react';
import Checkbox from './Checkbox';
import renderer from 'react-test-renderer';
//import { shallow } from 'enzyme';

describe('<Checkbox />', () => {

  let checked = false;
  let component;
  let tree;
  let reRender;
  const makeRenderer = component =>
    () => tree = component.toJSON();

  beforeEach(() => {
    component = renderer.create(
      <Checkbox
        onChange={_e => checked = !checked}
        checked={checked}
      />);
    reRender = makeRenderer(component);
    reRender();
  })

  it('match snapshot', () => {
    expect(tree).toMatchSnapshot();
  });


  it('is true after a click', () => {
    tree.props.onChange();
    reRender();
    expect(checked).toBe(true);
  });

  it('is false after two clicks', () => {
    tree.props.onChange();
    tree.props.onChange();
    reRender();
    expect(tree).toMatchSnapshot();
  });
});
