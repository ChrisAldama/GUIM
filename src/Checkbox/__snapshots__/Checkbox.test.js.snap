// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> is false after two clicks 1`] = `
<input
  checked={true}
  name={undefined}
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`<Checkbox /> match snapshot 1`] = `
<input
  checked={false}
  name={undefined}
  onChange={[Function]}
  type="checkbox"
/>
`;
